lib/ansible/modules/cloud/amazon/rds_tag.py

```diff
@@ -0,0 +1,223 @@
+  tags:
+    description: >-
+      a hash/dictionary of tags to add to the resource;
+      '{"key":"value"}' and '{"key":"value","key":"value"}'
+    required: true
+extends_documentation_fragment:
+    - aws
+    - ec2
+
+author: "Akane Morikawa (@cahlchang)"
+'''
+
+EXAMPLES = '''
+# Note: None of these examples set aws_access_key, aws_secret_key, or region.
+# It is assumed that their matching environment variables are set.
+
+- name: add tags to instance
+  rds_tag:
+    region: eu-west-1
+    instance_name: name-database
+    state: present
+    tags:
+      Name: environment
+      env: sandbox
+
+- name: remove tags to instance
+  rds_tag:
+    region: eu-west-1
+    instance_name: name-database
+    state: absent
+    tags:
+      Name: environment
+      env: sandbox
+'''
+
+RETURN = '''
+---
+changed:
+  description: Whether there was any change.
+  returned: always
+  type: bool
+  sample: true
+message:
+  description: Summary of module execution results.
+  returned: always
+  type: string
+  sample: Add tag to arn:aws:rds:eu-west-1:xxxxxxxxxxxx:db:db_name instance
+response_metadata:
+  description: Value returned by boto3.
+  returned: When not in dry run mode.
+  type: dict
+  sample:
+    http_headers:
+      content-length: xxx
+      content-type: text/xml
+      date: Mon, 02 Apr 2018 03:53:23 GMT
+      x-amzn-requestid: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+    http_status_code: 200
+    request_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+    retry_attempts: 0
+'''
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import (
+    ec2_argument_spec, camel_dict_to_snake_dict, compare_aws_tags,
+    ansible_dict_to_boto3_tag_list, boto3_tag_list_to_ansible_dict)
+
+try:
+    import botocore
+except ImportError:
+    pass  # handled by AnsibleAWSModule
+
+
+def add_rds_tags(module, client, tags_configured, tags_operate):
+    lst_compare = compare_aws_tags(tags_configured['tags'],
+                                   tags_operate,
+                                   False)
+    if 0 == len(lst_compare[0]):
+        module.exit_json(message='tags already exists.', changed=False)
+
+    if module.check_mode:
+        module.exit_json(message='check mode.', changed=True)
+
+    tags_add = lst_compare[0]
+    try:
+        response = client.add_tags_to_resource(
+            ResourceName=tags_configured['arn'],
+            Tags=ansible_dict_to_boto3_tag_list(tags_add))
+    except (botocore.exceptions.ClientError,
+            botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Couldn't add tag to %s"
+                             % tags_configured['arn'])
+
+    response['message'] = 'Add %s tag to %s instance' % (
+        tags_add,
+        tags_configured['arn'])
+
+    return response
+
+
+def remove_rds_tags(module, client, tags_configured, tags_operate):
+    tags_inter = set(tags_operate.keys()) & set(tags_configured['tags'].keys())
```

Comment:
compare_aws_tags returns the tags to add and the tags to remove. The tags to remove are in the format of a list of keys. You could call that once in main rather than handle it in each function (you'll need to pass True as the last argument to compare_aws_tags, so it returns the list to remove if state is absent).